src/main/java/com/ait/lienzo/client/widget/panel/mediators/PanelMediators.java

```diff
@@ -0,0 +1,176 @@
+    private MouseWheelZoomMediator zoomMediator;
+    private MousePanMediator panMediator;
+    private PanelPreviewMediator previewMediator;
+    private HandlerRegistration outHandler;
+
+    public static PanelMediators build(final LienzoBoundsPanel panel) {
+        return new PanelMediators().init(new Supplier<LienzoBoundsPanel>() {
+            @Override
+            public LienzoBoundsPanel get() {
+                return panel;
+            }
+        });
+    }
+
+    public PanelMediators init(final Supplier<LienzoBoundsPanel> panelSupplier) {
+        return init(panelSupplier,
+                    new Supplier<PanelPreviewMediator>() {
+                        @Override
+                        public PanelPreviewMediator get() {
+                            return PanelPreviewMediator.build((ScrollablePanel) panelSupplier.get());
+                        }
+                    });
+    }
+
+    public PanelMediators init(final Supplier<LienzoBoundsPanel> panelSupplier,
+                               final Supplier<PanelPreviewMediator> previewMediatorBuilder) {
+        this.panelSupplier = panelSupplier;
+        final LienzoBoundsPanel panel = panelSupplier.get();
+
+        final Viewport viewport = getViewport();
+
+        zoomMediator = new MouseWheelZoomMediator()
+                .setMinScale(MIN_SCALE)
+                .setMaxScale(MAX_SCALE)
+                .setZoomFactor(ZOOM_FACTOR);
+
+        panMediator = new MousePanMediator()
+                .setXConstrained(true)
+                .setYConstrained(true);
+
+        zoomMediator.setEnabled(false);
+        panMediator.setEnabled(false);
+
+        viewport.getMediators().push(zoomMediator);
+        viewport.getMediators().push(panMediator);
+
+        if (panel instanceof ScrollablePanel) {
+            previewMediator = previewMediatorBuilder.get();
+        }
+
+        outHandler = panel.addMouseOutHandler(new MouseOutHandler() {
+            @Override
+            public void onMouseOut(MouseOutEvent mouseOutEvent) {
+                disable();
+            }
+        });
+        return this;
+    }
+
+    public boolean enableZoomMediator() {
+        if (panMediator.isEnabled() || previewMediator.isEnabled()) {
+            return false;
+        }
+        zoomMediator.setEnabled(true);
+        return true;
+    }
+
+    public boolean enablePanMediator() {
+        if (zoomMediator.isEnabled() || previewMediator.isEnabled()) {
+            return false;
+        }
+        panMediator.setEnabled(true);
+        return true;
+    }
+
+    public boolean enablePreviewMediator() {
+        if (null != previewMediator) {
+            previewMediator.enable();
+            return true;
+        }
+        return false;
+    }
+
+    public void disable() {
+        disableMediator(zoomMediator);
+        disableMediator(panMediator);
+        if (null != previewMediator) {
+            previewMediator.disable();
+        }
```

Comment:
Don't we miss `outHandler`?

Reply:
`panelSupplier` and `previewMediator` as well.

Reply:
This is the `disable` method, which is expected to just disable the previously enabled mediators, but the panel (supplier) and the event handlers attached to it should be still valid and listening, so any of the mediators can be enabled again on the same instance of `PanelMediators`. The panel and event handlers are only removed once calling the `destroy` method. So I think it's fine as it is...